File: src/TaskOTime/TaskOTime.ViewModel/ViewModels/MasterDataViewModel.vb
Imports System.Windows
Imports TaskOTime.AppServer.Models
Imports TaskOTime.AppServer.Services
Imports TaskOTime.ViewModel.Views

Namespace ViewModels
    Public Class MasterDataViewModel
        Private ReadOnly _view As MasterDataWindow
        Private ReadOnly _tenantUsers As TenantUserViewModel
        Private ReadOnly _projects As ProjectViewModel
        Private ReadOnly _tasks As MasterTaskViewModel
        Private ReadOnly _collaboration As CollaborationViewModel

        Public Sub New(view As MasterDataWindow, tenant As TenantDto, userId As Guid,
                       admin As IAdminMasterDataService, users As IUserAdministrationService,
                       bookings As ITimeBookingService, tab As Integer)
            _view = view
            Dim store = New ServiceWorkspace(tenant, userId, admin, users, bookings)
          'Die controls wohnen im Viewmodel.dann ist code-behind komplet weg und das ist ja die trennung
            _tenantUsers = New TenantUserViewModel(view.TenantUserScreen, store)
            _projects = New ProjectViewModel(view.ProjectScreen, store)
            _tasks = New MasterTaskViewModel(view.TaskScreen, store)
            _collaboration = New CollaborationViewModel(view.CollaborationScreen, store)
            view.DataContext = Me
            view.WorkspaceTabs.SelectedIndex = tab
            AddHandler view.AboutButton.Click, Sub(sender, args)
                                                 MessageBox.Show(view, "Stammdatenverwaltung", "Task-o-Time")
                                             End Sub
        End Sub
    End Class
End Namespace

File: src/TaskOTime/TaskOTime.ViewModel/ViewModels/MasterDataViewModel.cs
using System;
using TaskOTime.AppServer.Models;
using TaskOTime.AppServer.Services;
using TaskOTime.ViewModel.Views;

namespace TaskOTime.ViewModel.ViewModels
{
    public class MasterDataViewModel
    {
        private readonly MasterDataWindow _view;
        private readonly TenantUserViewModel _tenantUsers;
        private readonly ProjectViewModel _projects;
        private readonly MasterTaskViewModel _tasks;
        private readonly CollaborationViewModel _collaboration;

        public MasterDataViewModel(MasterDataWindow view, TenantDto tenant, Guid userId, IAdminMasterDataService admin, IUserAdministrationService users, ITimeBookingService bookings, int tab)
        {
            _view = view;
            var store = new ServiceWorkspace(tenant, userId, admin, users, bookings);
            // Die controls wohnen im Viewmodel.dann ist code-behind komplet weg und das ist ja die trennung
            _tenantUsers = new TenantUserViewModel(view.TenantUserScreen, store);
            _projects = new ProjectViewModel(view.ProjectScreen, store);
            _tasks = new MasterTaskViewModel(view.TaskScreen, store);
            _collaboration = new CollaborationViewModel(view.CollaborationScreen, store);
            view.DataContext = this;
            view.WorkspaceTabs.SelectedIndex = tab;
            view.AboutButton.Click += (sender, args) => global::System.Windows.MessageBox.Show(view, "Stammdatenverwaltung", "Task-o-Time");
        }
    }
}